### PR TITLE
example: two crashed examples

### PR DESCRIPTION
**Pull request by SAutum**
# add two crashed examples in balance

Following up on issue #238  .

Proposed changes:
- add a 2d bowtie example in the _balance_ example
- add a 2d drop example in the _balance_ example
- some minor changes along the way to fix warnings in doc generation

### CODE REVIEW

**Comment by cburstedde** [comment]
Thanks; if you could add a line to the doc/release_notes.txt file, and another retroactively for your previous #236?

**Comment by cburstedde** [comment]
Thanks! I'm thinking that the 2D and 3D corner examples have much code in common. Do you think you could eliminate the 3.c file to the usual 2-liner we use in the library files and work with `#ifdef P4_TO_P8` in the 2.c example file?

**Comment by SAutum** [comment]
> Thanks! I'm thinking that the 2D and 3D corner examples have much code in common. Do you think you could eliminate the 3.c file to the usual 2-liner we use in the library files and work with `#ifdef P4_TO_P8` in the 2.c example file?

OK. Will do.

**Comment by cburstedde** [comment]
> > Thanks! I'm thinking that the 2D and 3D corner examples have much code in common. Do you think you could eliminate the 3.c file to the usual 2-liner we use in the library files and work with `#ifdef P4_TO_P8` in the 2.c example file?
> 
> OK. Will do.

Thanks; please note that I have pushed a small update to this branch in the meantime.

File: example/balance/balance_corner2.c [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+#include <p4est_bits.h>
+#include <p4est_extended.h>
+#include <p4est_vtk.h>
+#else
+#include <p8est_bits.h>
+#include <p8est_extended.h>
+#include <p8est_vtk.h>
+#endif
+
+typedef struct
```
Comment (cburstedde) [inline example/balance/balance_corner2.c:35]: This example contains some historic items for demonstration purposes.
The user_data may be deleted here, and data_size hardcoded as 0 below.

File: example/balance/balance_corner2.c [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  int                 mpirank;
+}
+mpi_context_t;
+
+/* refinement level initialization */
+static int          refine_level = 0;
+
+/* connectivity initialization function */
+static void
+init_fn (p4est_t * p4est, p4est_topidx_t which_tree,
```
Comment (cburstedde) [inline example/balance/balance_corner2.c:54]: This init function is for demonstration purposes and fine in itself.
This example is a good candidate for maximum simplification, so deleting init and hardcoding it below as NULL would be fine.

File: example/balance/balance_corner2.c [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  }
+
+  /* assign variables based on configuration */
+  refine_level = atoi (argv[1]);
+
+  /* create connectivity and forest structures */
+  geom = NULL;
+#ifndef P4_TO_P8
+  connectivity = p4est_connectivity_new_bowtie ();
+  p4est = p4est_new_ext (mpi->mpicomm, connectivity, 15, 0, 0,
```
Comment (cburstedde) [inline example/balance/balance_corner2.c:127]: The new_ext function is translated and can live outside the #ifdefs. We had historically chosen the parameter 15 for min_quadrants, but that makes the refinement partition-dependent, which we later found undesirable. This value can be set to 0. Conversely, the fill_uniform parameter would need to be set to 1 to make sure that the refinement looks the same independent of the number of cores used.

File: example/balance/balance_corner2.c [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+                         sizeof (user_data_t), init_fn, geom);
+#else
+  connectivity = p8est_connectivity_new_drop ();
+  p4est = p8est_new_ext (mpi->mpicomm, connectivity, 15, 0, 0,
+                         sizeof (user_data_t), init_fn, geom);
+#endif
+  p4est_vtk_write_file (p4est, geom, "test");
```
Comment (cburstedde) [inline example/balance/balance_corner2.c:134]: We may consistently produce output with file names, after calling the respective algorithms, like
P4EST_STRING "_corner_new",
P4EST_STRING "_corner_refine",
P4EST_STRING "_corner_balance",
which will allow us to write the VTK outside the dimension #ifdefs.

Comment (SAutum) [reply example/balance/balance_corner2.c:134]: Deleted. The statement serves a test on the p4-to-p8 functionality.

Comment (cburstedde) [reply example/balance/balance_corner2.c]: I don't mind writing a VTK after new, after refine, after balance, and after partition.
With the name convention as above, this should all work without conflict:
with P4EST_STRING, the vtk_write_file function can be called outside the #ifdef.

**Comment by cburstedde** [comment]
Thanks, I have pushed on top as discussed in our video meeting.

**Pull request merged**